doc/source/train/examples/lightning/lightning_cola_advanced.ipynb

```diff
@@ -217,75 +216,92 @@
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 12,
+   "execution_count": 7,
    "metadata": {},
    "outputs": [],
    "source": [
-    "from ray.train import RunConfig, ScalingConfig, CheckpointConfig\n",
-    "from ray.train.lightning import LightningTrainer, LightningConfigBuilder\n",
+    "import ray.train\n",
+    "from ray.train.lightning import (\n",
+    "    prepare_trainer,\n",
+    "    RayDDPStrategy,\n",
+    "    RayLightningEnvironment,\n",
+    "    RayTrainReportCallback,\n",
+    ")\n",
     "\n",
-    "# Define the configs for LightningTrainer\n",
-    "lightning_config = (\n",
-    "    LightningConfigBuilder()\n",
-    "    .module(cls=SentimentModel, lr=1e-5, eps=1e-8)\n",
-    "    .trainer(max_epochs=5, accelerator=\"gpu\")\n",
-    "    .checkpointing(save_on_train_epoch_end=False)\n",
-    "    .build()\n",
-    ")"
+    "train_func_config = {\n",
+    "    \"lr\": 1e-5,\n",
+    "    \"eps\": 1e-8,\n",
+    "    \"batch_size\": 16,\n",
+    "    \"max_epochs\": 5,\n",
+    "}\n",
+    "\n",
+    "def train_func(config):\n",
+    "    # Unpack the input configs passed from `TorchTrainer(train_loop_config)`\n",
+    "    lr = config[\"lr\"]\n",
+    "    eps = config[\"eps\"]\n",
+    "    batch_size = config[\"batch_size\"]\n",
+    "    max_epochs = config[\"max_epochs\"]\n",
+    "\n",
+    "    # Fetch the Dataset shards\n",
+    "    train_ds = ray.train.get_dataset_shard(\"train\")\n",
+    "    val_ds = ray.train.get_dataset_shard(\"validation\")\n",
+    "\n",
+    "    # Create a dataloader for Ray Datasets\n",
+    "    train_ds_loader = train_ds.iter_torch_batches(batch_size=batch_size)\n",
+    "    val_ds_loader = val_ds.iter_torch_batches(batch_size=batch_size)\n",
+    "\n",
+    "    # Model\n",
+    "    model = SentimentModel(lr=lr, eps=eps)\n",
+    "\n",
+    "    trainer = pl.Trainer(\n",
+    "        max_epochs=max_epochs,\n",
+    "        accelerator=\"auto\",\n",
+    "        devices=\"auto\",\n",
+    "        strategy=RayDDPStrategy(),\n",
+    "        plugins=[RayLightningEnvironment()],\n",
+    "        callbacks=[RayTrainReportCallback()],\n",
+    "        enable_progress_bar=False,\n",
+    "    )\n",
+    "\n",
+    "    trainer = prepare_trainer(trainer)\n",
+    "\n",
+    "    trainer.fit(model, train_dataloaders=train_ds_loader, val_dataloaders=val_ds_loader)\n"
    ]
   },
   {
-   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    ":::{note}\n",
-    "Note that the `lightning_config` is created on the head node and will be passed to the worker nodes later. Be aware that the environment variables and hardware settings may differ between the head node and worker nodes.\n",
-    ":::\n",
+    "As discussed in {ref}`Lightning Quick Start Guide <train-pytorch-lightning>`, to enable distributed training with Ray Train, we need to inject several utilities to the lightning's trainer.\n",
```

Comment:
For some reason it doesn't let me put a diff here, but I think we can simplify this to:

```
To enable distributed training with Ray Train, configure the Lightning Trainer with the following utilities:
```

We can also move the quick start reference into the `seealso` below, which can then contain the quick start and the data guide.